tests/unit/test_client_logging.py

```diff
@@ -0,0 +1,17 @@
+import logging
+import pytest
+
+from google.api_core.client_logging import BaseLogger
+
```

Comment:
I suggest adding some cases testing `setup_logging` with no scopes, the `google` scope, a `google.foo` scope, and an `random.bar` scope, and then testing the expected behavior: warnings, propagation, handler+formatter. WDYT?

Reply:
Re-opening as we should document/test the formatter that we expect in each of the test cases

Reply:
The formatter in each of the test case is structured format. I think testing anything from the default config is sufficient to confirm that they're applied. Let me know what you think!

Reply:
@ohmayr : OK, that sounds fine. Separately, though, it would be useful to have a test for the formatter, where we call `StructuredLogFormatter.format(sample_data)` directly and verify we have the JSON output as expected. It would be a sort of golden test for the formatter (field names and values populated correctly). We can do it as a follow-up. (Maybe add a TODO with an issue link in this file?)

Reply:
addressed.